docs/access_token-pop.md

```diff
@@ -0,0 +1,178 @@
+
+   Proof of Possession (PoP) increases the security posture of these tokens embedding them inside of a JWT envelope and signing (binding) that JWT with RSA key material. The key material is generated on the device which was originally issued the tokens and never leaves it. The resulting JWT is called the Signed HTTP Request (SHR). This binding renders the access token unusable without a recent proof for the target resource endpoint.
+
+2. **Basic Pop Token Flow.**
+
+   ```
+   +--------+                               +---------------+
+   |        |--(A)- Authorization Request ->|   Resource    |
+   |        |                               |     Owner     |
+   |        |<-(B)-- Authorization Grant ---|               |
+   |        |                               +---------------+
+   |        |
+   |        |                               +---------------+
+   |        |--(C)-- Authorization Grant -->|               |
+   | Client |       (resource, req_cnf)     | Authorization |
+   |        |                               |     Server    |
+   |        |<-(D)-- PoP Access Token ------|               |
+   |        |       (rs_cnf, token_type)    +---------------+
+   |        |        
+   |        |                               +---------------+
+   |        |--(E)-- PoP Access Token ----->|               |
+   |        |   (with proof of private key) |    Resource   |
+   |        |                               |     Server    |
+   |        |<-(F)--- Protected Resource ---|               |
+   +--------+                               +---------------+
+   ```
+
+   2.1	In the Token Request, the client sends an authorization grant, e.g., an authorization code or a refresh token, to the authorization server in order to obtain an access token (and potentially a refresh token). The client proves the possession of a private key belonging to some public key by first generating an asymmetric RSA key pair on the device. The client then sends an additional parameter "req_cnf" to the token endpoint which contains a thumbprint of the RSA public key the client would like to associate with the access token. Please refer to public key thumbprint computation spec here [https://tools.ietf.org/id/draft-ietf-oauth-pop-key-distribution-04.html]()
+
+   2.2	The AS binds (sender-constrains) the access token to the public key claimed by the client; that is, the access token cannot be used without proving possession of the respective private key. This is signaled to the client by using the token_type value "pop" and by appending a "cnf" claim in the access token jwt containing a "kid" member identifying the public key.
```

Comment:
What is AS here? (might be worth explaining it in docs)

Reply:
AS is short hand for Authorization Server. Added explanation for that.